includes/admin/wc-admin-functions.php

```diff
@@ -251,6 +251,9 @@ function wc_save_order_items( $order_id, $items ) {
 				}
 			}
 
+			// Allow other plugins to change item object before it is saved.
+			do_action( 'woocommerce_before_save_order_item', $item, $item_data, $order_id );
```

Comment:
Is item data and order id needed? Doesn't the item already contain these?

Reply:
Yes, $item already contain these, but usually related data is still passed to filters/actions. For example in add_order_item() function, you can find this filter:
```
$item    = apply_filters( 'woocommerce_ajax_order_item', $order->get_item( $item_id ), $item_id );
```
So even if $item has of course its ID, $item_id is still passed to that filter.

Reply:
If it's not necessary I wouldn't include it. If we do need to add args in the future not accessible via $item, this makes it more difficult.

Reply:
Do you think that this would be enough then?
```
do_action( 'woocommerce_before_save_order_item', $item, $order_id );
```

Reply:
I don't even think order_id is needed - it's stored within the item.